centreon/www/include/monitoring/recurrentDowntime/formDowntime.php

```diff
@@ -40,6 +40,60 @@
     exit();
 }
 
+/**
+ * GetUserAclAllowedResources returns allowed resources for the user regarding resource type
+ *
+ * @param CentreonACL $userAcl
+ * @param string $resourceType
+ */
+function getUserAclAllowedResources(CentreonACL $userAcl, string $resourceType)
+{
+    return match ($resourceType) {
+        'hosts' => $userAcl->getHostAclConf(null, 'broker'),
+        'servicegroups' => $userAcl->getServiceGroupAclConf(null, 'broker'),
+        'hostgroups' => $userAcl->getHostGroupAclConf(null, 'broker'),
+    };
+}
+
+/**
+ * Check resources access regarding selected resources
+ *
+ * @param CentreonACL $userAcl
+ * @param int[] $selectedResources
+ * @param string $resourceType
+ * @return bool
+ **/
+function checkResourcesRelations(CentreonACL $userAcl, array $selectedResources, string $resourceType): bool
+{
+    $allowedResources = getUserAclAllowedResources($userAcl, $resourceType);
+
+    $selectedResourceIds = array_map(
+        function (array $resource) {
+            return $resource['id'];
+        },
+        $selectedResources
+    );
+
+    $diff = array_diff($selectedResourceIds, array_keys($allowedResources));
+
+    if (empty($diff)) {
+        return true;
+    }
+
+    /**
+     * checking that remain diff is not filled with disabled hosts
+     */
+    foreach ($selectedResources as $resource) {
+        if (
+            in_array($resource['id'], $diff)
+            && $resource['activated'] === '0'
+        ) {
+            continue;
+        }
+    }
```

Comment:
This foreach does nothing : forgotten `return false` ?

Reply:
You are absolutely right